app/src/main/java/org/wikipedia/LongPressHandler.java

```diff
@@ -55,67 +50,64 @@ public void onCreateContextMenu(ContextMenu menu, View view,
                     WikiSite wikiSite = new WikiSite(uri);
                     // the following logic keeps the correct language code if the domain has multiple variants (e.g. zh).
-                    if (wikiSite.dbName().equals(((WebViewContextMenuListener) contextMenuListener).getWikiSite().dbName())
-                            && !wikiSite.languageCode().equals(((WebViewContextMenuListener) contextMenuListener).getWikiSite().languageCode())) {
-                        wikiSite = ((WebViewContextMenuListener) contextMenuListener).getWikiSite();
+                    if (wikiSite.dbName().equals(((WebViewDialogMenuListener) dialogMenuListener).getWikiSite().dbName())
+                            && !wikiSite.languageCode().equals(((WebViewDialogMenuListener) dialogMenuListener).getWikiSite().languageCode())) {
+                        wikiSite = ((WebViewDialogMenuListener) dialogMenuListener).getWikiSite();
                     }
                     title = wikiSite.titleForInternalLink(uri.getPath());
                 }
             }
         } else if (view instanceof ListView) {
-            AdapterView.AdapterContextMenuInfo info = (AdapterView.AdapterContextMenuInfo) menuInfo;
-            title = ((ListViewContextMenuListener) contextMenuListener)
-                    .getTitleForListPosition(info.position);
+            title = ((ListViewDialogMenuListener) dialogMenuListener).getTitleForListPosition(((ListView) view).getSelectedItemPosition());
         }
 
         if (title != null && !title.isSpecial()) {
             hideSoftKeyboard(view);
             entry = new HistoryEntry(title, historySource);
             entry.setReferrer(referrer);
-            new MenuInflater(view.getContext()).inflate(R.menu.menu_page_long_press, menu);
-            menu.setHeaderTitle(title.getDisplayText());
-            for (int i = 0; i < menu.size(); i++) {
-                menu.getItem(i).setOnMenuItemClickListener(this);
-            }
-        }
-    }
 
-    @Override
-    public boolean onMenuItemClick(MenuItem item) {
-        switch (item.getItemId()) {
-            case R.id.menu_long_press_open_page:
-                contextMenuListener.onOpenLink(title, entry);
-                return true;
-            case R.id.menu_long_press_open_in_new_tab:
-                contextMenuListener.onOpenInNewTab(title, entry);
-                return true;
-            case R.id.menu_long_press_copy_page:
-                contextMenuListener.onCopyLink(title);
-                return true;
-            case R.id.menu_long_press_share_page:
-                contextMenuListener.onShareLink(title);
-                return true;
-            case R.id.menu_long_press_add_to_list:
-                contextMenuListener.onAddToList(title, CONTEXT_MENU);
-                return true;
-            default:
-            return false;
+            AlertDialog.Builder builder = new AlertDialog.Builder(view.getContext());
```

Comment:
Can we use a `PopupMenu` instead of `AlertDialog`? That way, we can directly inflate the menu resource, and not require another array of strings.

Reply:
Sure, it would be much easier. Only concern is that we are no longer able to set the header for the menu.